examples/navier-stokes/densitycurrent.h

```diff
@@ -24,12 +24,49 @@
 #ifndef densitycurrent_h
 #define densitycurrent_h
 
-#include <math.h>
+#ifndef __CUDACC__
+#  include <math.h>
+#endif
 
 #ifndef M_PI
 #define M_PI    3.14159265358979323846
 #endif
 
+#ifndef setup_context_struct
+#define setup_context_struct
+typedef struct SetupContext_ *SetupContext;
+struct SetupContext_ {
+  CeedScalar theta0;
+  CeedScalar thetaC;
+  CeedScalar P0;
+  CeedScalar N;
+  CeedScalar cv;
+  CeedScalar cp;
+  CeedScalar Rd;
+  CeedScalar g;
+  CeedScalar rc;
+  CeedScalar lx;
+  CeedScalar ly;
+  CeedScalar lz;
+  CeedScalar periodicity0;
+  CeedScalar periodicity1;
+  CeedScalar periodicity2;
+  CeedScalar center[3];
+  CeedScalar dc_axis[3];
+  CeedScalar time;
+};
+#endif
+
+#ifndef advection_context_struct
+#define advection_context_struct
+typedef struct AdvectionContext_ *AdvectionContext;
+struct AdvectionContext_ {
+  CeedScalar CtauS;
+  CeedScalar strong_form;
+  int stabilization; // See StabilizationType: 0=none, 1=SU, 2=SUPG
+};
+#endif
```

Comment:
Yuck, but I see why it's needed at present. I have sights on refactoring to make it easier to define new configurations (initial/boundary conditions and quantities of interest), but I think that'll have to be after v0.6, which Tzanio would like released this week.

Reply:
I'm not wild about the guarded multiple definitions either but nothing better comes to mind with our current infrastructure.